### PR TITLE
[new release] flint (4 packages) (0.3.1)

### PR DESCRIPTION
**Pull request by bobot**
Stub of the C library Flint2

- Project page: <a href="https://github.com/bobot/ocaml-flint">https://github.com/bobot/ocaml-flint</a>

##### CHANGES:

  - Use installed C libraries instead of compiling them internally

### CODE REVIEW

File: packages/conf-calcium/conf-calcium.1/opam [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		"-exc"
+		"$(ocamlc -config-var c_compiler) -c $CFLAGS -I/usr/local/include test.c"
+	] {os = "win32" & os-distribution = "cygwinports"}
+  [
+    "sh"
+    "-exc"
+    "cc -c $CFLAGS -I/opt/homebrew/include -I/opt/local/include -I/usr/local/include test.c"
+  ] {os = "macos"}
+]
+depexts: [
```
Comment (mseri) [inline packages/conf-calcium/conf-calcium.1/opam:21]:
```suggestion
depends: [
  "conf-flint"
]
depexts: [
```
due to
```
#=== ERROR while compiling conf-calcium.1 =====================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.09.1 | pinned
# path                 ~/.opam/4.09/.opam-switch/build/conf-calcium.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build sh -exc cc -c $CFLAGS -I/usr/local/include test.c
# exit-code            1
# env-file             ~/.opam/log/conf-calcium-7-b5c1a2.env
# output-file          ~/.opam/log/conf-calcium-7-b5c1a2.out
### output ###
# + cc -c -I/usr/local/include test.c
# In file included from test.c:1:
# /usr/include/calcium/calcium.h:22:10: fatal error: flint/flint.h: No such file or directory
#    22 | #include "flint/flint.h"
#       |          ^~~~~~~~~~~~~~~
# compilation terminated.
```

File: packages/conf-calcium/conf-calcium.1/opam [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		"$(ocamlc -config-var c_compiler) -c $CFLAGS -I/usr/local/include test.c"
+	] {os = "win32" & os-distribution = "cygwinports"}
+  [
+    "sh"
+    "-exc"
+    "cc -c $CFLAGS -I/opt/homebrew/include -I/opt/local/include -I/usr/local/include test.c"
+  ] {os = "macos"}
+]
+depends: [
+  "conf-flint"
```
Comment (mseri) [inline packages/conf-calcium/conf-calcium.1/opam:22]:
```suggestion
  "conf-arb"
  "conf-flint"
```
Provides acb.h

File: packages/antic/antic.0.3.1/opam [hunk 3]
```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+synopsis: "Stub of the C library Antic. Algebraic number"
+maintainer: ["François Bobot"]
+authors: ["François Bobot"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/bobot/ocaml-flint"
+bug-reports: "https://github.com/bobot/ocaml-flint/issues"
+depends: [
+  "dune" {>= "3.2"}
```
Comment (mseri) [inline packages/antic/antic.0.3.1/opam:9]:
```suggestion
  "dune" {>= "3.2" & < "3.11.0"}
```
Due to
```
#=== ERROR while compiling flint.0.3.1 ========================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.12.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.12/.opam-switch/build/flint.0.3.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p flint -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/flint-7-281834.env
# output-file          ~/.opam/log/flint-7-281834.out
### output ###
# File "dune-project", line 4, characters 14-17:
# 4 | (using ctypes 0.1)
#                   ^^^
# Error: Version 0.1 of the ctypes extension has been deleted in Dune 3.11.
# Please port this project to a newer version of the extension, such as 0.3.
# Hint: You will also need to upgrade to (lang dune 3.7).
```

File: packages/arb/arb.0.3.1/opam [hunk 4]
```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+synopsis: "Stub of the C library Arb. Ball approximation"
+maintainer: ["François Bobot"]
+authors: ["François Bobot"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/bobot/ocaml-flint"
+bug-reports: "https://github.com/bobot/ocaml-flint/issues"
+depends: [
+  "dune" {>= "3.2"}
```
Comment (mseri) [inline packages/arb/arb.0.3.1/opam:9]:
```suggestion
  "dune" {>= "3.2" & < "3.11.0"}
```

File: packages/calcium/calcium.0.3.1/opam [hunk 5]
```diff
@@ -0,0 +1,51 @@
+opam-version: "2.0"
+synopsis:
+  "Stub of the C library Antic. For exact computation with real and complex numbers, presently in early development"
+maintainer: ["François Bobot"]
+authors: ["François Bobot"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/bobot/ocaml-flint"
+bug-reports: "https://github.com/bobot/ocaml-flint/issues"
+depends: [
+  "dune" {>= "3.2"}
```
Comment (mseri) [inline packages/calcium/calcium.0.3.1/opam:10]:
```suggestion
  "dune" {>= "3.2" & < "3.11.0"}
```

File: packages/flint/flint.0.3.1/opam [hunk 6]
```diff
@@ -0,0 +1,47 @@
+opam-version: "2.0"
+synopsis: "Stub of the C library Flint2"
+maintainer: ["François Bobot"]
+authors: ["François Bobot"]
+license: "LGPL-2.1-only"
+homepage: "https://github.com/bobot/ocaml-flint"
+bug-reports: "https://github.com/bobot/ocaml-flint/issues"
+depends: [
+  "dune" {>= "3.2"}
```
Comment (mseri) [inline packages/flint/flint.0.3.1/opam:9]:
```suggestion
  "dune" {>= "3.2" & < "3.11.0"}
```

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

(Note the upper bound on dune, @bobot -- this will be problematic in the long term so you may want to release a newer version that fixes that)

**Pull request merged**